src/scss/foundations/typography/index.scss

```diff
@@ -0,0 +1,242 @@
+      lib.rem(lib.getOrDie($_letterSpacing, medium))
+    );
+  }
+}
+
+/**
+ * Include the font-weight value by name
+ *
+ * @param {String} $weight - The name of the weight
+ * @throws {Error}
+ *
+ * @access private
+ */
+@mixin _fontWeight($weight) {
+  font-weight: #{lib.getOrDie($_font, weights, $weight)};
+}
+
+/**
+ * Includes the properties contained in the typography map in every breakpoints
+ *
+ * @param {Map} $fontSizes - The map of fontSize
+ * @param {String} $size - Name of the $size
+ * @throws {Error}
+ *
+ * @access private
+ */
+@mixin _responsiveTypographySize($fontSizes, $size) {
+  @each $breakpoint, $value in lib.getOrDie($fontSizes, $size) {
+    @include breakpoints.mq($breakpoint) {
+      $fontSize: lib.getOrDie($fontSizes, $size, $breakpoint);
+
+      font-size: lib.rem($fontSize);
+      letter-spacing: _letterSpacing($fontSize);
+      line-height: _lineHeight($fontSize);
+    }
+  }
+}
+
+/**
+ * Include the responsive title size by name available in title.$fontSizes
+ *
+ * @param {String} $size - The name of the size
+ * @param {String} $weight - The name of the weight
+ * @throws {Error}
+ *
+ * @access public
+ */
+@mixin title($size, $weight: bold) {
+  @include _responsiveTypographySize(title.$fontSizes, $size);
+
+  @if index(title.$allowedFontWeights, $weight) {
+    @include _fontWeight($weight);
+  } @else {
+    @error "Font-weight `#{$weight}` is not allowed in titles";
+  }
+}
+
+/**
+ * Include the responsive text size by name available in text.$fontSizes
+ *
+ * @param {String} $size - The name of the size
+ * @param {String} $weight - The name of the weight
+ * @throws {Error}
+ *
+ * @access public
+ */
+@mixin text($size, $weight: book) {
+  @include _fontWeight($weight);
+  @include _responsiveTypographySize(text.$fontSizes, $size);
+}
+
+/**
+ * Base setting for Pyxis typography
+ * To be used only in the html tag
+ *
+ * @access public
+ */
+@mixin typographySetting() {
+  @include _fontWeight(book);
+  font-family: _fontFamily($_font);
+  font-feature-settings: lib.getOrDie($_font, featureSettings);
+  font-size: global.$fontSizeRoot;
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-font-smoothing: antialiased;
+  font-variant-ligatures: none;
+  text-rendering: optimizeLegibility;
+}
+
+/**
+ * Create atom classes for all title and text variant
+ *
+ * @access public
+ */
+@mixin createAtoms() {
+  @each $size, $_ in title.$fontSizes {
+    @each $weight in title.$allowedFontWeights {
+      .title-#{lib.camelToKebabCase($size)}--#{$weight} {
+        @include title($size, $weight);
+      }
+    }
+  }
+
+  @each $size, $_ in text.$fontSizes {
+    @each $weight in text.$allowedFontWeights {
+      .text-#{lib.camelToKebabCase($size)}--#{$weight} {
+        @include text($size, $weight);
+      }
+    }
+  }
+}
+
+/**
+ * Create @font-face for all fonts available in $_fonts map
+ */
+@each $name, $weight in lib.getOrDie($_font, weights) {
```

Comment:
Questo lo metterei in un `@mixin fontFaces` da invocarsi direttamente in fondo al file, cioè prima degli atoms che devono rimanere sempre l'ultima funzione in fondo.

Tutto ciò solo per non avere cicli e snippets volanti

Reply:
Ok fatto il mixin e invocato a fondo file, così abbiamo come ordine finale: variabili, funzioni, mixin, invocazioni